src/aker_metrics.c

```diff
@@ -0,0 +1,202 @@
+	aker_metrics_t * tmp = NULL;
+	tmp = g_metrics;
+
+	va_list valist;
+	va_start(valist, num);
+
+	switch(metrics)
+	{
+		case 0:
+			tmp->device_block_count += va_arg(valist, uint32_t);
+			va_end(valist);
+			debug_print("After changing tmp->device_block_count %d\n", tmp->device_block_count);
+			break;
+
+		case 1:
+			tmp->windows_transistion_count += va_arg(valist, uint32_t);
+			va_end(valist);
+			debug_print("After changing tmp->windows_transistion_count %d\n", tmp->windows_transistion_count);
+			break;
+
+		case 2:
+			tmp->schedule_set_count += va_arg(valist, uint32_t);
+			va_end(valist);
+			debug_print("After changing tmp->schedule_set_count %d\n", tmp->schedule_set_count);
+			break;
+
+		case 3:
+			tmp->md5_error_count += va_arg(valist, uint32_t);
+			va_end(valist);
+			debug_print("After changing tmp->md5_error_count %d\n", tmp->md5_error_count);
+			break;
+
+		case 4:
+			tmp->process_start_time = va_arg(valist, time_t);
+			va_end(valist);
+			debug_print("After changing tmp->process_start_time %ld\n", (long)tmp->process_start_time);
+			break;
+
+		case 5:
+			tmp->schedule_enabled = va_arg(valist, int);
+			va_end(valist);
+			debug_print("After changing tmp->schedule_enabled %d\n", tmp->schedule_enabled);
+			break;
+
+		case 6:
+			tmp->timezone = NULL;
+			tmp->timezone = strdup(va_arg(valist, char*));
+			va_end(valist);
+			debug_print("After changing tmp->timezone %d\n", tmp->timezone);
+			break;
+
+		default:
+			debug_error("The mentioned metrics is not found\n");
+			return 0;
+	}
+	return 1;	
+}
+
+int stringify_metrics(int flag)
+{
+	char str[512];
+	aker_metrics_t * tmp = NULL;
+	tmp = g_metrics;
+
+	if(g_metrics != NULL && flag == 1)
+	{
+		snprintf(str, 512, "%s%d%s%d%s%d%s%d%s%ld%s%d%s%s",
+						"DeviceBlockCount,", tmp->device_block_count,
+                                                  ",WindowTransistionCount,", tmp->windows_transistion_count,
+                                                    ",ScheduleSetCount,", tmp->schedule_set_count,
+                                                    ",MD5ErrorCount,", tmp->md5_error_count,
+                                                    ",ProcessStartTime,", tmp->process_start_time,
+                                                    ",ScheduleEnabled,", tmp->schedule_enabled,
+                                                    ",TimeZone,", tmp->timezone);
+
+		debug_info("The stringified valued is (%s)\n", str);
+
+#if defined(ENABLE_FEATURE_TELEMETRY2_0)
+		t2_event_s("akermetric", str);
+		debug_info("akermetric t2 event triggered\n");
+#endif
+
+	}
+	else
+	{
+		if(flag == 0)
+		{
+			debug_info("The current stringified value is (DeviceBlockCount, %d, WindowTransistionCount, %d, ScheduleSetCount, %d, MD5ErrorCount, %d, ProcessStartTime, %ld, ScheduleEnabled, %d, TimeZone, %s)\n", tmp->device_block_count, tmp->windows_transistion_count, tmp->schedule_set_count, tmp->md5_error_count, tmp->process_start_time, tmp->schedule_enabled, tmp->timezone);
```

Comment:
Why wouldn't you use the stringified value from above here vs. duplicating the print statement?